tests/codegen/optimize-attr-1.rs

```diff
@@ -15,6 +15,17 @@ pub fn nothing() -> i32 {
     2 + 2
 }
 
+// CHECK-LABEL: define{{.*}}i32 @none
+// CHECK-SAME: [[NONE_ATTRS:#[0-9]+]]
+// SIZE-OPT: alloca
+// SPEED-OPT: alloca
+#[no_mangle]
+#[optimize(none)]
+pub fn none() -> i32 {
+    let arr = [0, 1, 2, 3, 4];
+    arr[4]
+}
```

Comment:
in addition to this, I'd also like to see some mir-opt tests. For example, that mir passes don't optimize the Add operand out of `1 + 2` (if that is indeed the case)